packages/turf-rewind/index.ts

```diff
@@ -21,25 +33,24 @@ import { featureCollection, isObject } from "@turf/helpers";
  * //addToMap
  * var addToMap = [rewind];
  */
-function rewind(geojson, options) {
+function rewind<T extends AllGeoJSON>(
+  geojson: T,
+  options: {
+    reverse?: boolean;
+    mutate?: boolean;
+  } = {}
+): Geometry | Feature | FeatureCollection {
   // Optional parameters
   options = options || {};
   if (!isObject(options)) throw new Error("options is invalid");
-  var reverse = options.reverse || false;
-  var mutate = options.mutate || false;
+  const mutate = options.mutate || false;
+  const reverse = options.reverse || false;
 
-  // validation
-  if (!geojson) throw new Error("<geojson> is required");
-  if (typeof reverse !== "boolean")
-    throw new Error("<reverse> must be a boolean");
-  if (typeof mutate !== "boolean")
-    throw new Error("<mutate> must be a boolean");
-
-  // prevent input mutation
-  if (mutate === false) geojson = clone(geojson);
+  // Prevent input mutation if requested.
+  if (!mutate) geojson = clone(geojson);
```

Comment:
Possibly a little sketchy but we can skip this whole copy for Point and MultiPoint geometries. Or maybe just don't even accept them into the function in the first place.

Reply:
Have added some logic to only clone if not Point or MultiPoint.

Reply:
The thing that's sketchy is that if you use rewind assuming that its then safe to mutate the return because the rewind cloned it, you suddenly have different behavior on Point and MultiPoint which is confusing. Sorry we maybe should just always clone.